packages/stack/compiler/src/index.ts

```diff
@@ -17,53 +21,131 @@ class Compiler {
+            const osTmpDir = os.tmpdir();
 
-    async.eachObject(this.getAvailableCompilers(),
-      (extension: string, compilers: any, next: any) => {
-        const matchingFiles = contractFiles.filter(this.filesMatchingExtension(extension));
-        if (matchingFiles.length === 0) {
-          return next();
-        }
+            for (const file of contractFilesArr) {
+              if (file instanceof File) {
+                _contractFilesArr.push(file);
+                continue;
+              }
 
-        async.someLimit(compilers, 1, (compiler: any, someCb: Callback<boolean>) => {
-          compiler.call(compiler, matchingFiles, compilerOptions, (err: any, compileResult: any) => {
-            if (err) {
-              return someCb(err);
+              if (!file.path) {
+                err = new TypeError("path property was missing on contract file object");
+                break;
+              }
+
+              let filePath: string;
+              if (!path.isAbsolute(file.path)) {
+                filePath = dappPath(file.path);
+              } else {
+                filePath = path.normalize(file.path);
+              }
+
+              if (![_dappPath, osTmpDir].some((dir) => filePath.startsWith(dir))) {
+                err = new Error("path must be within the DApp project or the OS temp directory");
+                break;
+              }
+
+              _contractFilesArr.push(
+                new File({
+                  originalPath: file.path,
+                  path: file.path,
+                  resolver: (callback: Callback<any>) => {
+                    fs.readFile(file.path, { encoding: "utf-8" }, callback);
+                  },
+                  type: Types.dappFile,
+                }),
+              );
             }
-            if (compileResult === false) {
-              // Compiler not compatible, trying the next one
-              return someCb(null, false);
+
+            if (err) {
+              return next(err);
             }
-            Object.assign(compiledObject, compileResult);
-            someCb(null, true);
+
+            next(null, _contractFilesArr);
           });
-        }, (err: Error, result: boolean) => {
-          if (err) {
-            return next(err);
-          }
-          if (!result) {
-            // No compiler was compatible
-            return next(new Error(__("No installed compiler was compatible with your version of %s files", extension)));
-          }
-          next();
-        });
-      },
-      (err: any) => {
-        contractFiles.filter((f: any) => !f.compiled).forEach((file: any) => {
-          this.logger.warn(__("%s doesn't have a compatible contract compiler. Maybe a plugin exists for it.", file.path));
-        });
+        },
+        (contractFilesArr: any[], next: Callback<any>) => {
+          const compiledObject: { [index: string]: any } = {};
 
-        cb(err, compiledObject);
+          const compilerOptions = {
```

Comment:
This isn't necessary anymore and can be done away with.

Reply:
Nice, thanks.

Reply:
Done. Though, for the moment I have:

```
compilerOptions = {};
```

An options object is passed to compilers, e.g. `embark-solc`, but it seems that we never passed any options from this script except `isCoverage`.

Reply:
Yeah this is from when we did coverage by removing compiler optimization and map out the bytecode back to the original source.

Good times.